qiskit/transpiler/passes/synthesis/hls_plugins.py

```diff
@@ -1029,3 +1147,129 @@ def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **
             high_level_object.num_target_qubits,
             ctrl_state,
         )
+
+
+class ModularAdderSynthesisDefault(HighLevelSynthesisPlugin):
+    """The default modular adder (no carry in, no carry out qubit) synthesis."""
+
+    def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **options):
+        if options.get("num_clean_ancillas", 0) >= 1:
```

Comment:
Should we, in this and other plugins, check that the type of the high level object is really an adder, to prevent users from defining custom gates with names "Adder", "FullAdder" or "ModularAdder". I am especially worried about the more generic name "Adder".

Reply:
Could you please document the option "num_clean_ancillas" in the plugin's docstring? And also to explicitly say what is the key (e.g. "ripple_c04") with which the plugin will be called? 

This comment also applies to other plugins.

Reply:
Done in a54907eec954f98f61bf723fe0e82d5f4d0fb00e